src/models/subscription.ts

```diff
@@ -50,6 +50,15 @@ export class Subscription extends Model {
 		});
 	}
 
+	static async getAllForHostAndCloud(jiraHost: string): Promise<Subscription[]> {
```

Comment:
Isn't it better to just modify the existing `getAllForHost`? 
If we change the condition in `where` to do add the `gitHubAppId` if its not undefined, then it should work. 
For cloud, we pass null values: `.getAllForHost(jiraHost, null)`, which makes sense cause cloud apps have `null` `gitHubAppId`
And for enterprise, we pass number: `getAllForHost(jiraHost, numberVal)`.

This way we won't need the if-else check in the `GithubCreateBranchGet`. We can just do `await Subscription.getAllForHost(jiraHost, gitHubAppConfig.gitHubAppId || null)`.

```suggestion
					...(gitHubAppId !== undefined && { gitHubAppId }), // Add gitHubAppId only if passed

```

Reply:
`getAllForHost` is being used at multiple places. If we change modify the existing implementation, it may impact other flows.

Reply:
This change won't affect any other places. 
Originally, the second parameter is just another filter to fetch the list of subscriptions for server apps. So, the only use cases for this method so far is one of these two:
- `getAllForHost(jiraHost)` - without the second param, which means its `undefined`.
- `getAllForHost(jiraHost, id)` - with second param, where its a number.

What I'm suggesting is to avoid adding the `gitHubAppId` parameter only if it is `undefined`.  Basically separating `undefined` and `null` cases.

So, for the current case, we need to fetch the subscriptions for cloud or ghe server.

And if we get the subscriptions using: 
`await Subscription.getAllForHost(jiraHost, gitHubAppConfig.gitHubAppId || null)`

Then this will work for both cases:
- For cloud it will pass in null value and the filter becomes  `{ gitHubAppId: null, jiraHost }` 
- And for enterprise, it will pass in a number value and the filter becomes `{ gitHubAppId, jiraHost }`.

And for the already used ones, nothing's really changed.